ScheduledExecutorServiceBasicTest.schedule_withNamedInstanceAware_whenRemoteRun

http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-3.maintenance-IbmJDK7/109/testReport/com.hazelcast.scheduledexecutor/ScheduledExecutorServiceBasicTest/schedule_withNamedInstanceAware_whenRemoteRun/

**Stacktrace**
```
java.lang.AssertionError: expected:<0> but was:<1>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.scheduledexecutor.ScheduledExecutorServiceBasicTest.schedule_withNamedInstanceAware_whenRemoteRun(ScheduledExecutorServiceBasicTest.java:799)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:95)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:56)
	at java.lang.reflect.Method.invoke(Method.java:620)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:274)
	at java.lang.Thread.run(Thread.java:809)
```


```
Hiccups measured while running test 'schedule_withNamedInstanceAware_whenRemoteRun(com.hazelcast.scheduledexecutor.ScheduledExecutorServiceBasicTest):'
03:00:45, accumulated pauses: 190 ms, max pause: 162 ms, pauses over 1000 ms: 0
03:00:50, accumulated pauses: 1002 ms, max pause: 980 ms, pauses over 1000 ms: 0
03:00:55, accumulated pauses: 62 ms, max pause: 4 ms, pauses over 1000 ms: 0
03:01:00, accumulated pauses: 106 ms, max pause: 20 ms, pauses over 1000 ms: 0
03:01:05, accumulated pauses: 85 ms, max pause: 6 ms, pauses over 1000 ms: 0
03:01:10, accumulated pauses: 1008 ms, max pause: 934 ms, pauses over 1000 ms: 0
03:01:15, accumulated pauses: 52 ms, max pause: 5 ms, pauses over 1000 ms: 0
03:01:20, accumulated pauses: 53 ms, max pause: 6 ms, pauses over 1000 ms: 0
03:01:25, accumulated pauses: 21 ms, max pause: 6 ms, pauses over 1000 ms: 0
```

--- a/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceBasicTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceBasicTest.java
@@ -48,7 +48,6 @@ import java.util.concurrent.Callable;
 import java.util.concurrent.CancellationException;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.RejectedExecutionException;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
 import static com.hazelcast.scheduledexecutor.TaskUtils.named;
@@ -100,13 +99,12 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
     }
 
     @Test
-    public void exception_suppressesFutureExecutions()
-            throws ExecutionException, InterruptedException {
+    public void exception_suppressesFutureExecutions() throws Exception {
         HazelcastInstance[] instances = createClusterWithCount(2);
         IScheduledExecutorService service = instances[0].getScheduledExecutorService("test");
 
         final IScheduledFuture f = service.scheduleAtFixedRate(
-                new ErroneousRunnableTask(), 1, 1, TimeUnit.SECONDS);
+                new ErroneousRunnableTask(), 1, 1, SECONDS);
 
         assertTrueEventually(new AssertTask() {
             @Override
@@ -139,7 +137,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         String keyOwner = "hitSamePartitionToCheckCapacity";
 
         for (int i = 0; i < 101; i++) {
-            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, TimeUnit.SECONDS);
+            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, SECONDS);
         }
     }
 
@@ -152,11 +150,11 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         String keyOwner = "hitSamePartitionToCheckCapacity";
 
         for (int i = 0; i < 100; i++) {
-            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, TimeUnit.SECONDS);
+            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, SECONDS);
         }
 
         try {
-            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, TimeUnit.SECONDS);
+            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, SECONDS);
             fail("Should have been rejected.");
         } catch (RejectedExecutionException ex) {
             assertEquals("Got wrong RejectedExecutionException",
@@ -182,11 +180,11 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         String keyOwner = "hitSamePartitionToCheckCapacity";
 
         for (int i = 0; i < 10; i++) {
-            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, TimeUnit.SECONDS);
+            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, SECONDS);
         }
 
         try {
-            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, TimeUnit.SECONDS);
+            service.scheduleOnKeyOwner(new PlainCallableTask(), keyOwner, 0, SECONDS);
             fail("Should have been rejected.");
         } catch (RejectedExecutionException ex) {
             assertEquals("Got wrong RejectedExecutionException",
@@ -212,11 +210,11 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         Member member = instances[0].getCluster().getLocalMember();
 
         for (int i = 0; i < 10; i++) {
-            service.scheduleOnMember(new PlainCallableTask(), member, 0, TimeUnit.SECONDS);
+            service.scheduleOnMember(new PlainCallableTask(), member, 0, SECONDS);
         }
 
         try {
-            service.scheduleOnMember(new PlainCallableTask(), member, 0, TimeUnit.SECONDS);
+            service.scheduleOnMember(new PlainCallableTask(), member, 0, SECONDS);
             fail("Should have been rejected.");
         } catch (RejectedExecutionException ex) {
             assertEquals("Got wrong RejectedExecutionException",
@@ -243,7 +241,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
     }
 
     @Test
-    public void handlerTaskAndSchedulerNames_withRunnable() throws Exception {
+    public void handlerTaskAndSchedulerNames_withRunnable() {
         int delay = 0;
         String schedulerName = "s";
         String taskName = "TestRunnable";
@@ -257,7 +255,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         IScheduledFuture future = executorService.schedule(
                 named(taskName, new ICountdownLatchRunnableTask("latch")), delay, SECONDS);
 
-        latch.await(10, SECONDS);
+        assertOpenEventually(latch);
 
         ScheduledTaskHandler handler = future.getHandler();
         assertEquals(schedulerName, handler.getSchedulerName());
@@ -343,7 +341,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
     }
 
     @Test
-    public void schedule_withMapChanges_durable() throws Exception {
+    public void schedule_withMapChanges_durable() {
         HazelcastInstance[] instances = createClusterWithCount(2);
         IMap<String, Integer> map = instances[1].getMap("map");
         for (int i = 0; i < MAP_INCREMENT_TASK_MAX_ENTRIES; i++) {
@@ -375,7 +373,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
     }
 
     @Test
-    public void schedule_withLongSleepingCallable_cancelledAndGet() throws InterruptedException {
+    public void schedule_withLongSleepingCallable_cancelledAndGet() {
         int delay = 0;
 
         HazelcastInstance[] instances = createClusterWithCount(2);
@@ -389,7 +387,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         sleepSeconds(4);
         future.cancel(false);
 
-        runsCountLatch.await(15, SECONDS);
+        assertOpenEventually(runsCountLatch);
 
         assertTrue(future.isDone());
         assertTrue(future.isCancelled());
@@ -457,7 +455,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         IScheduledExecutorService executorService = getScheduledExecutor(instances, "s");
         IScheduledFuture<Double> first = executorService.schedule(named(taskName, new PlainCallableTask()), delay, MINUTES);
 
-        first.get(2, TimeUnit.SECONDS);
+        first.get(2, SECONDS);
     }
 
     @Test
@@ -738,7 +736,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
     }
 
     @Test
-    public void schedule_partitionAware_runnable() throws Exception {
+    public void schedule_partitionAware_runnable() {
         int delay = 1;
         String completionLatchName = "completionLatch";
 
@@ -750,7 +748,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         Runnable task = new PlainPartitionAwareRunnableTask(completionLatchName);
         IScheduledFuture first = executorService.schedule(task, delay, SECONDS);
 
-        completionLatch.await(10, SECONDS);
+        assertOpenEventually(completionLatch);
 
         ScheduledTaskHandler handler = first.getHandler();
         int expectedPartition = getPartitionIdFromPartitionAwareTask(instances[0], (PartitionAware) task);
@@ -758,7 +756,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
     }
 
     @Test
-    public void schedule_withStatefulRunnable() throws Exception {
+    public void schedule_withStatefulRunnable() {
         HazelcastInstance[] instances = createClusterWithCount(4);
         IScheduledExecutorService executorService = getScheduledExecutor(instances, "s");
 
@@ -767,40 +765,38 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
 
         executorService.schedule(new StatefulRunnableTask("latch", "runC", "loadC"), 2, SECONDS);
 
-        latch.await(10, SECONDS);
+        assertOpenEventually(latch);
     }
 
     @Test
-    public void schedule_withNamedInstanceAware_whenLocalRun()
-            throws InterruptedException {
+    public void schedule_withNamedInstanceAware_whenLocalRun() {
         HazelcastInstance[] instances = createClusterWithCount(1);
         ICountDownLatch latch = instances[0].getCountDownLatch("latch");
         latch.trySetCount(1);
 
         IScheduledExecutorService s = getScheduledExecutor(instances, "s");
-        s.schedule(TaskUtils.named("blah", new PlainInstanceAwareRunnableTask("latch")), 1, TimeUnit.SECONDS);
+        s.schedule(TaskUtils.named("blah", new PlainInstanceAwareRunnableTask("latch")), 1, SECONDS);
 
-        latch.await(20, SECONDS);
+        assertOpenEventually(latch);
         assertEquals(0, latch.getCount());
     }
 
     @Test
-    public void schedule_withNamedInstanceAware_whenRemoteRun()
-            throws InterruptedException {
+    public void schedule_withNamedInstanceAware_whenRemoteRun() {
         HazelcastInstance[] instances = createClusterWithCount(2);
         ICountDownLatch latch = instances[0].getCountDownLatch("latch");
         latch.trySetCount(1);
 
         MemberImpl member = getNodeEngineImpl(instances[1]).getLocalMember();
         IScheduledExecutorService s = getScheduledExecutor(instances, "s");
-        s.scheduleOnMember(TaskUtils.named("blah", new PlainInstanceAwareRunnableTask("latch")), member, 1, TimeUnit.SECONDS);
+        s.scheduleOnMember(TaskUtils.named("blah", new PlainInstanceAwareRunnableTask("latch")), member, 1, SECONDS);
 
-        latch.await(20, SECONDS);
+        assertOpenEventually(latch);
         assertEquals(0, latch.getCount());
     }
 
     @Test
-    public void scheduleWithRepetition() throws Exception {
+    public void scheduleWithRepetition() {
         HazelcastInstance[] instances = createClusterWithCount(2);
 
         IScheduledExecutorService s = getScheduledExecutor(instances, "s");
@@ -810,7 +806,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
 
         IScheduledFuture future = s.scheduleAtFixedRate(new ICountdownLatchRunnableTask("latch"), 0, 1, SECONDS);
 
-        latch.await(20, SECONDS);
+        assertOpenEventually(latch);
         future.cancel(false);
 
         assertEquals(0, latch.getCount());
@@ -831,7 +827,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
     }
 
     @Test
-    public void scheduleOnMemberWithRepetition() throws Exception {
+    public void scheduleOnMemberWithRepetition() {
         HazelcastInstance[] instances = createClusterWithCount(4);
         IScheduledExecutorService s = getScheduledExecutor(instances, "s");
 
@@ -841,7 +837,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         Map<Member, IScheduledFuture<?>> futures = s.scheduleOnAllMembersAtFixedRate(
                 new ICountdownLatchRunnableTask("latch"), 0, 3, SECONDS);
 
-        latch.await(10, SECONDS);
+        assertOpenEventually(latch);
 
         assertEquals(0, latch.getCount());
         assertEquals(4, futures.size());
@@ -908,7 +904,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
     }
 
     @Test
-    public void scheduleOnKeyOwnerWithRepetition() throws Exception {
+    public void scheduleOnKeyOwnerWithRepetition() {
         String key = "TestKey";
 
         HazelcastInstance[] instances = createClusterWithCount(2);
@@ -927,7 +923,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
 
         assertEquals(expectedPartition, handler.getPartitionId());
 
-        latch.await(60, SECONDS);
+        assertOpenEventually(latch);
         assertEquals(0, latch.getCount());
     }
 
@@ -1004,7 +1000,7 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         IScheduledFuture<Double> future = executorService.scheduleOnKeyOwner(
                 named(taskName, new ErroneousCallableTask(completionLatchName)), key, delay, SECONDS);
 
-        latch.await(10, SECONDS);
+        assertOpenEventually(latch);
         expected.expect(ExecutionException.class);
         expected.expectCause(new RootCauseMatcher(IllegalStateException.class, "Erroneous task"));
         future.get();
@@ -1026,9 +1022,9 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
         IScheduledFuture<Double> future = executorService.scheduleOnKeyOwner(
                 named(taskName, new ErroneousCallableTask(completionLatchName)), key, delay, SECONDS);
 
-        latch.await(10, SECONDS);
+        assertOpenEventually(latch);
         instances[1].getLifecycleService().shutdown();
-        Thread.sleep(2000);
+        sleepAtLeastSeconds(2);
 
         expected.expect(ExecutionException.class);
         expected.expectCause(new RootCauseMatcher(IllegalStateException.class, "Erroneous task"));

--- a/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java
@@ -82,15 +82,16 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
                 key, 10, 10, SECONDS);
 
         // wait for task to get scheduled and start
-        latch.await(11, SECONDS);
+        assertOpenEventually(latch);
 
-        Thread.sleep(waitStateSyncPeriodToAvoidPassiveState);
+        sleepAtLeastMillis(waitStateSyncPeriodToAvoidPassiveState);
 
         instances[1].getLifecycleService().shutdown();
 
         // reset latch - task should be running on a replica now
         latch.trySetCount(7);
-        latch.await(70, SECONDS);
+
+        assertOpenEventually(latch);
         future.cancel(false);
 
         assertEquals(getPartitionService(instances[0]).getPartitionId(key), future.getHandler().getPartitionId());
@@ -114,11 +115,12 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
         IScheduledFuture future = executorService.scheduleOnKeyOwnerAtFixedRate(
                 new ICountdownLatchRunnableTask("firstLatch", "lastLatch"), key, 0, 10, SECONDS);
 
-        firstLatch.await(12, SECONDS);
+        assertOpenEventually(firstLatch);
 
         instances[1].getLifecycleService().shutdown();
 
-        lastLatch.await(70, SECONDS);
+        assertOpenEventually(lastLatch);
+
         // wait for run-cycle to finish before cancelling, in order for stats to get updated
         sleepSeconds(4);
         future.cancel(false);
@@ -138,7 +140,7 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
         IScheduledFuture future = executorService.scheduleAtFixedRate(
                 new ICountdownLatchRunnableTask("latch"), 0, 10, SECONDS);
 
-        latch.await(120, SECONDS);
+        assertOpenEventually(latch);
         future.cancel(false);
 
         ScheduledTaskStatistics stats = future.getStats();
@@ -190,7 +192,7 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
             s.scheduleOnKeyOwnerAtFixedRate(new ICountdownLatchRunnableTask(runsCounterName), key, 0, 2, SECONDS);
         }
 
-        runsLatch.await(10, SECONDS);
+        assertOpenEventually(runsLatch);
         instances[1].getLifecycleService().shutdown();
 
         assertEquals(expectedTotal, countScheduledTasksOn(s), 0);
@@ -217,7 +219,7 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
             }
         }
 
-        runsLatch.await(10, SECONDS);
+        assertOpenEventually(runsLatch);
 
         int actualTotal = 0;
         for (int i = 0; i < numOfSchedulers; i++) {
@@ -249,7 +251,7 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
             }
         }
 
-        runsLatch.await(10, SECONDS);
+        assertOpenEventually(runsLatch);
 
         getScheduledExecutor(instances, "scheduler_" + 0).shutdown();
         getScheduledExecutor(instances, "scheduler_" + 1).shutdown();
@@ -284,7 +286,7 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
             }
         }
 
-        runsLatch.await(10, SECONDS);
+        assertOpenEventually(runsLatch);
 
         instances[1].getLifecycleService().terminate();
 
@@ -316,7 +318,7 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
         assertTrue(future.isDone());
 
         // wait till the task is actually done, since even though we cancelled the task is current task is still running
-        latch.await(60, SECONDS);
+        assertOpenEventually(latch);
 
         // make sure SyncState goes through
         sleepSeconds(10);
@@ -349,7 +351,7 @@ public class ScheduledExecutorServiceSlowTest extends ScheduledExecutorServiceTe
         assertTrue(future.isDone());
 
         // wait till the task is actually done, since even though we cancelled the task is current task is still running
-        latch.await(60, SECONDS);
+        assertOpenEventually(latch);
 
         // make sure SyncState goes through
         sleepSeconds(10);

--- a/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceTestSupport.java
+++ b/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceTestSupport.java
@@ -32,7 +32,6 @@ import java.util.Map;
 import java.util.concurrent.Callable;
 
 import static java.lang.System.currentTimeMillis;
-import static java.lang.Thread.sleep;
 
 /**
  * Common methods used in ScheduledExecutorService tests.
@@ -125,11 +124,7 @@ public class ScheduledExecutorServiceTestSupport extends HazelcastTestSupport {
 
         @Override
         public Double call() {
-            try {
-                sleep(sleepPeriod);
-            } catch (InterruptedException e) {
-                Thread.interrupted();
-            }
+            sleepAtLeastMillis(sleepPeriod);
 
             instance.getCountDownLatch(runLatchName).countDown();
             return 77 * 2.2;
@@ -216,14 +211,10 @@ public class ScheduledExecutorServiceTestSupport extends HazelcastTestSupport {
         public void run() {
             long start = currentTimeMillis();
             while (true) {
-                try {
-                    sleep(5000);
-                    if (currentTimeMillis() - start >= 30000) {
-                        instance.getCountDownLatch(runFinishedLatchName).countDown();
-                        break;
-                    }
-                } catch (InterruptedException e) {
-                    // ignore
+                sleepAtLeastMillis(5000);
+                if (currentTimeMillis() - start >= 30000) {
+                    instance.getCountDownLatch(runFinishedLatchName).countDown();
+                    break;
                 }
             }
         }

